client_integration_test.go

```diff
@@ -2444,6 +2454,28 @@ func BenchmarkReadFrom4MiBDelay150Msec(b *testing.B) {
 	benchmarkReadFrom(b, 4*1024*1024, 150*time.Millisecond)
 }
 
+// writeToBuffer implements the relevant parts of bytes.Buffer,
+// but does not release its internal buffer when Reset.
+//
+// Release its internal memory when Reset is good for avoiding memory leaks,
+// but not great for memory benchmarks, as this fills up a lot of irrelevant allocations.
+type writeToBuffer struct {
```

Comment:
:thinking: WTF? If I look at `bytes.Buffer.Reset` it keeps its backing buffer the same as we do here.

Yet, without this, I was seeing significant memory allocation in `bytes.Buffer.Grow()` :thinking:

Reply:
Yep, looks like it was growing from somewhere else entirely.